feat: granular aztec js

Thunkar opened the pull request:
Crude approach towards a granular `@aztec/aztec.js`. Makes it much more tree-shakable and tidy in general. It needs some work to move some exports around, but at least the big barrel export is gone, which will make for a brighter future.

nventuro commented on boxes/boxes/vanilla/.env:
```diff
@@ -1,4 +1,4 @@
+CONTRACT_ADDRESS=0x3003bd5fe49b41a31115f2aefbfca522587b5ebc129023378719b3acba852d56
+DEPLOYER_ADDRESS=0x15c4d593bdbde3c2fa73f2d525c9b6a8fe3bece0c402a90aa8df863ffe59654b
+DEPLOYMENT_SALT=0x1040425a765c001830de9ad8236e62d365b7edde814b07bff4a2a9322c3933c9
+AZTEC_NODE_URL=http://aztec:8080
```
Looks like this shnouldn't be committed?

Thunkar replied:
Yup, good catch

nventuro reviewed (approved):
Lovely. Missing migration notes though - feel free to merge once those are added. We don't need to show where every type went, just tell people 'hey this changed, you'll get a smaller bundler, delete the imports and have your IDE re-do them'.

Thunkar committed:
feat: granular aztec js

Crude approach towards a granular `@aztec/aztec.js`. Makes it much more tree-shakable and tidy in general. It needs some work to move some exports around, but at least the big barrel export is gone, which will make for a brighter future.
